### DISCUSSION TITLE
Hyprland on RISC-V

### DISCUSSION THREAD

**Discussion by twenex78** [hyprwm/Hyprland#4613] [General]
I've recently acquired a RISC-V development board (the Lichee Pi 4A - I'm typing on it now) and noticed that Hyprland is not available. I downloaded the sources for Hyprland and a few of its dependencies, but getting it to work on RISC-V is biting off more than I can chew.

I would love it if someone could port Hyprland to RISC-V. 

So far I've found that in order to get Hyprland to work on the board, you need:

1. Python
2. Ninja
(Both installable but not installed by default on the system's default image)
3. libpango and libpango cairo (available for RISC-V in the regular debian source repositories for RISC-, but this is where I ran into problems)
4. wayland- server, -client and -cursor. I've looked in the debian sources, but they don't seem to be discrete packages in there. (wayland-protocols is available in the revyos fork which is the default installation source for Debian as supplied with the board.)

I haven't looked for xkbcommon, libinput or pixman-1 in the (official) debian sources, but they're probably there or will be soon, since RISC-V has been a supported architecture on sid for nearly a year now.

**Comment by vaxerski** [comment]
I don't think there is much to "port", you just have to compile a lot of stuff from source, gentoo style.

**Reply by twenex78** [reply]
Fair enough, I will look at this again.

**Reply by MrDuartePT** [reply]
Gentoo ebuild is keyword for `~riscv` (~ just mean the ebuild is unstable ) this mean  hyprland was tested on riscv hardware, so it should compile properly if you have all the dependencies.

Note: The ebuild being unstable dosent mean Hyprland not work well, it just riscv is quite “new” and there isn’t much comercial available hardware to test like amd64 ou arm
